assets/src/onboarding-wizard/index.js

```diff
@@ -90,13 +95,24 @@ Providers.propTypes = {
 domReady( () => {
 	const root = document.getElementById( APP_ROOT_ID );
 
-	if ( root ) {
-		render(
-
-			<Providers>
-				<SetupWizard closeLink={ CLOSE_LINK } finishLink={ FINISH_LINK } appRoot={ root } />
-			</Providers>,
-			root,
-		);
+	if ( ! root ) {
+		return;
 	}
+
+	const errorHandler = ( error ) => {
+		// Handle only own errors.
+		if ( error?.filename?.match( /amp-onboarding-wizard(\.min)?\.js/ ) ) {
+			render( <ErrorScreen error={ error } />, root );
```

Comment:
The error event is passed here, instead of the error object, so this should actually be:

```suggestion
			render( <ErrorScreen error={ error.error } />, root );
```